src/asynqp/routing.py

```diff
@@ -41,6 +59,27 @@ def notify(self, method, result=None):
         fut.set_result(result)
         self._futures.remove_item(fut)
 
+    def notify_connection_closed(self):
+        self.connection_closed = True
+        # Give a proper notification to blocking methods
+        for method in self._blocking_methods:
+            while True:
+                try:
+                    self.notify(method)
+                except StopIteration:
+                    break
+
+        # Set an exception for all others
+        for method in self._futures.keys():
+            if method not in self._blocking_methods:
+                while True:
+                    try:
+                        fut = self._futures.get_leftmost(method)
+                        fut.set_exception(ConnectionError)
```

Comment:
Just spotted this - should this be fut.set_exception(ConnectionError())? Without the extra () you're just passing a reference to the class. (Or does set_exception try to instantiate the exception for you?)